src/client/runner/automation/playback/drag.js

```diff
@@ -1,209 +1,197 @@
-                offsetY: pointTo.y
-            };
-
-            var moveOptions = new MoveOptions();
-
-            moveOptions.offsetX       = offsetsTo.offsetX;
-            moveOptions.offsetY       = offsetsTo.offsetY;
-            moveOptions.modifiers     = modifiers;
-            moveOptions.speed         = DRAGGING_SPEED;
-            moveOptions.minMovingTime = MIN_MOVING_TIME;
-            moveOptions.dragMode      = true;
-
-            var moveAutomation = new MoveAutomation(targetTo, moveOptions);
-
-            moveAutomation
-                .run()
-                .then(() => window.setTimeout(callback, automationSettings.DRAG_ACTION_STEP_DELAY));
-        },
-
-
-        cursorMouseUp: function (callback) {
-            if (pointTo)
-                screenPointTo = positionUtils.offsetToClientCoords(pointTo);
-            else {
-                var offsetPos = positionUtils.getOffsetPosition(el);
-
-                screenPointTo = positionUtils.offsetToClientCoords({
-                    x: offsetPos.left,
-                    y: offsetPos.top
-                });
-            }
-
-            eventPointTo = automationUtil.getEventOptionCoordinates(el, screenPointTo);
-
-            if (domUtils.isElementInIframe(el)) {
-                var currentIFrame = domUtils.getIframeByElement(el);
-                if (currentIFrame) {
-                    var screenPointToInIFrame = {
-                        x: screenPointTo.x - styleUtils.getScrollLeft(currentIFrame.contentWindow),
-                        y: screenPointTo.y - styleUtils.getScrollTop(currentIFrame.contentWindow)
-                    };
-
-                    topElement = getElementFromPoint(screenPointToInIFrame.x, screenPointToInIFrame.y);
-                }
-            }
-            else
-                topElement = getElementFromPoint(screenPointTo.x, screenPointTo.y);
-
-            if (!topElement) {
-                runCallback();
-                return;
-            }
-
-            eventOptionsEnd = extend({
-                clientX: eventPointTo.x,
-                clientY: eventPointTo.y
-            }, options);
-
-            cursor
-                .buttonUp()
-                .then(() => callback());
-        },
-
-        mouseUp: function (callback) {
-            if (browserUtils.hasTouchEvents)
-                eventSimulator.touchend(topElement, eventOptionsEnd);
-            else
-                eventSimulator.mouseup(topElement, eventOptionsEnd);
-
-            callback();
-        },
-
-        click: function () {
-            //B231323
-            if (getElementFromPoint(screenPointTo.x, screenPointTo.y) === topElement)
-                eventSimulator.click(topElement, eventOptionsEnd);
-
-            runCallback();
-        }
-    });
-};
+    _focus () {
+        return new Promise(resolve => {
+            // NOTE: For contentEditable elements we should call focus directly for action's element because
```

Comment:
If a target element is a child of a contentEditable element, we need to call focus for the contentEditable element.